app/controllers/users_controller.rb

```diff
@@ -1,13 +1,17 @@
 class UsersController < ApplicationController
   before_action :set_user, only: %i[ show liked feed followers following discover ]
+  before_action {authorize @user || User}
```

Comment:
great! You can use similar before actions across all of your controllers. This is more concise.